File: Shell_Sort.java
/*Implement Shell sort using Java*/
import java.util.*;
public class Shell
{
	public static void main(String abcd[])
	{

		Scanner s=new Scanner(System.in);
		int[] a=new int[30];
		System.out.println("Enter the size of array");
		int n=s.nextInt();
		System.out.println("Enter the array elements");
		for(int i=0;i<n;i++)
		{
			a[i]=s.nextInt();
		}
		for(int i=n/2;i>0;i=i/2)
		{
			for(int j=i;j<n;j++)
			{
				for(int k=j-i;k>=0;k=k-i)
				{
					if(a[k]>a[k+i])
					{
						int temp=a[k];
						a[k]=a[k+i];
						a[k+i]=temp;
					}
				
				}				
			}
		}
		for(int i=0;i<n;i++)
		{
			System.out.print(a[i]+"\t");
		}
	}
}	

File: Shell_Sort.cpp
/*Implement Shell Sort Using cpp*/
#include<iostream>
using namespace std;
int main()
{
	int a[10],n,i,j,temp,k;
	cout<<"Enter the number of elements";
	cin>>n;
	cout<<"Enter the elements";
	for(i=0;i<n;i++)
	{
		cin>>a[i];	
	}
	for(i=n/2;i>0;i=i/2)
	{
		for(j=i;j<n;j++)
		{
			for(k=j-i;k>=0;k=k-i)
			{
				if(a[k]>a[k+i])
				{
					temp=a[k];
					a[k]=a[k+i];
					a[k+i]=temp;
				}
			}		
		}
	}
	cout<<"Array after sorting";
	for(i=0;i<n;i++)
	{
		cout<<a[i];	
	}
	return 0;
}
